serving/src/main/java/feast/serving/controller/ServingServiceGRpcController.java

```diff
@@ -92,6 +101,10 @@ public void getBatchFeatures(
       GetBatchFeaturesRequest request, StreamObserver<GetBatchFeaturesResponse> responseObserver) {
     try {
       RequestHelper.validateBatchRequest(request);
+      // Authorize only valid request, Get the project from 1st feature.
+      // "Project" remains same in the request for all features.
+      this.authorizationService.authorizeRequest(
+          SecurityContextHolder.getContext(), request.getFeatures(0).getProject());
```

Comment:
See comment above about **multiple projects** second case

Reply:
this needs safe handling regardless.